pkg/handlers/config.go

```diff
@@ -229,47 +235,46 @@ func updateAppConfig(updateApp *app.App, seq int, req UpdateAppConfigRequest, is
 		return updateAppConfigResponse, err
 	}
 
-	err = updateApp.RenderDir(archiveDir)
+	registrySettings, err := registry.GetRegistrySettingsForApp(updateApp.ID)
+	if err != nil {
+		updateAppConfigResponse.Error = "failed to get registry settings"
+		return updateAppConfigResponse, err
+	}
+	err = render.RenderDir(archiveDir, updateApp.ID, registrySettings)
 	if err != nil {
 		updateAppConfigResponse.Error = "failed to render archive directory"
 		return updateAppConfigResponse, err
 	}
 
 	if req.CreateNewVersion {
-		newSequence, err := updateApp.CreateVersion(archiveDir, "Config Change")
+		newSequence, err := version.CreateVersion(updateApp.ID, archiveDir, "Config Change", updateApp.CurrentSequence)
 		if err != nil {
 			updateAppConfigResponse.Error = "failed to create an app version"
 			return updateAppConfigResponse, err
 		}
+		sequence = newSequence
 
-		if err := app.CreateAppVersionArchive(updateApp.ID, newSequence, archiveDir); err != nil {
+		if err := version.CreateAppVersionArchive(updateApp.ID, newSequence, archiveDir); err != nil {
 			updateAppConfigResponse.Error = "failed to create an app version archive"
 			return updateAppConfigResponse, err
 		}
 	} else {
-		if err := app.UpdateConfigValuesInDB(archiveDir, updateApp.ID, int64(seq)); err != nil {
+		if err := config.UpdateConfigValuesInDB(archiveDir, updateApp.ID, int64(sequence)); err != nil {
 			updateAppConfigResponse.Error = "failed to update config values in db"
 			return updateAppConfigResponse, err
 		}
 
-		if kotsKinds.Preflight != nil {
-			if err := app.SetDownstreamVersionPendingPreflight(updateApp.ID, int64(seq)); err != nil {
-				updateAppConfigResponse.Error = "failed to set downstream version pending preflight"
-				return updateAppConfigResponse, err
-			}
-		} else {
-			if err := app.SetDownstreamVersionReady(updateApp.ID, int64(seq)); err != nil {
-				updateAppConfigResponse.Error = "failed to set downstream version ready"
-				return updateAppConfigResponse, err
-			}
-		}
-
-		if err := app.CreateAppVersionArchive(updateApp.ID, int64(seq), archiveDir); err != nil {
+		if err := version.CreateAppVersionArchive(updateApp.ID, int64(sequence), archiveDir); err != nil {
 			updateAppConfigResponse.Error = "failed to create app version archive"
 			return updateAppConfigResponse, err
 		}
 	}
 
+	if err := preflight.Run(updateApp.ID, int64(sequence), archiveDir); err != nil {
```

Comment:
i guess we'll see if this is better than running them in a loop.  I think this breaks the "retry on missing RBAC" logic.